app/src/main/java/nodomain/freeyourgadget/gadgetbridge/util/FileUtils.java

```diff
@@ -111,15 +117,23 @@ public static void copyURItoFile(Context ctx, Uri uri, File destFile) throws IOE
         }
     }
 
+    /**
+     * Copies the content of a file to an uri,
+     * which for example was retrieved using the storage access framework.
+     * @param context the application context.
+     * @param src the file from which the content should be copied.
+     * @param dst the destination uri.
+     * @throws IOException
+     */
     public static void copyFileToURI(Context context, File src, Uri dst) throws IOException {
         OutputStream out = context.getContentResolver().openOutputStream(dst);
         if (out == null) {
             throw new IOException("Unable to open output stream for " + dst.toString());
         }
         try (OutputStream bufOut = new BufferedOutputStream(out)) {
             copyFileToStream(src, bufOut);
-            bufOut.close();
         }
+        out.close();
```

Comment:
This is also not necessary -- closing the `BufferedOutputStream` using the try-with-resources statement will automatically close the wrapped `out` stream as well.